backend/capellacollab/core/logging.py

```diff
@@ -0,0 +1,155 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import logging.handlers
+import os
+import random
+import string
+import typing as t
+
+from fastapi import Request, Response
+from starlette.middleware.base import BaseHTTPMiddleware
+
+from capellacollab.config import config
+from capellacollab.core.authentication.helper import get_username
+from capellacollab.core.authentication.jwt_bearer import JWTBearer
+
+LOGGING_LEVEL = config["logging"]["level"]
+
+
+class MakeTimedRotatingFileHandler(logging.handlers.TimedRotatingFileHandler):
+    def __init__(self, filename):
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+        super().__init__(filename, when="D", backupCount=1, delay=True)
+
+
+class AttachTraceIdMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        trace_id = "".join(
+            random.choices(string.ascii_uppercase + string.digits, k=6)
+        )
+        request.state.trace_id = trace_id
```

Comment:
Nice! Here I also wondered if we really "only" want to have a `trace_id` of length 6. It might be a good idea to increase it already to 10 or higher, so that we reduce the probability of getting duplicates.